src/bot/music.rs

```diff
@@ -556,7 +608,72 @@ impl Handler<MusicBotMessage> for MusicBot {
     }
 }
 
-fn spawn_stdin_reader(addr: WeakAddress<MusicBot>) {
+fn metadata_from_file(path: &Path, user: &str) -> Option<AudioMetadata> {
+    match path.extension().and_then(|s| s.to_str()) {
+        Some("mp3") => {
+            let tag = id3::Tag::read_from_path(&path).ok()?;
+            let title = match (tag.title(), tag.artist()) {
+                (Some(title), Some(artist)) => format!("{} - {}", title, artist),
+                (Some(title), _) => title.to_owned(),
+                (_, _) => path.file_name().unwrap().to_string_lossy().to_string(),
+            };
+
+            let mut cover = None;
+            for picture in tag.pictures() {
+                if picture.picture_type == id3::frame::PictureType::CoverFront {
+                    cover = Some(format!(
+                        "data:image/jpg;base64,{}",
+                        base64::encode(&picture.data)
```

Comment:
always jpeg?

Reply:
I guess not but browsers will probably show it correctly either way, so do we really want to try and figure out what the correct format is first if we don't know?